Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynAnalyzerProviderTests.cs

```diff
@@ -230,147 +175,157 @@ private List<SonarRule> CreateRules() =>
         </Rules>
         */
         [
-            new SonarRule("csharpsquid", "S1116", true)
+            new("csharpsquid", "S1116", true)
             {
-                Parameters = new Dictionary<string, string> { { "key", "value" } }
+                Parameters = new Dictionary<string, string> { { "key", "value" } },
             },
-            new SonarRule("csharpsquid", "S1125", true),
-            new SonarRule("roslyn.wintellect", "Wintellect003", true),
-            new SonarRule("csharpsquid", "S1000", false)
+            new("csharpsquid", "S1125", true),
+            new("roslyn.wintellect", "Wintellect003", true),
+            new("csharpsquid", "S1000", false)
         ];
 
-    private string CreateTestFolders()
-    {
-        var rootFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
-
-        // Create the binary and conf folders that are created by the bootstrapper
-        Directory.CreateDirectory(GetBinaryPath(rootFolder));
-        Directory.CreateDirectory(GetConfPath(rootFolder));
-
-        return rootFolder;
-    }
-
-    private static RoslynAnalyzerProvider CreateTestSubject(ILogger logger) =>
-        new(new MockAnalyzerInstaller(), logger);
-
     private static BuildSettings CreateSettings(string rootDir) =>
         BuildSettings.CreateNonTeamBuildSettingsForTesting(rootDir);
 
-    private static string GetConfPath(string rootDir) =>
-        Path.Combine(rootDir, "conf");
+    private class RoslynAnalyzerTestContext
+    {
+        public TestContext TestContext { get; set; }
```

Comment:
Can be private field
```suggestion
        private readonly TestContext testContext;
```